src/clr-angular/forms/select/select-container.ts

```diff
@@ -0,0 +1,81 @@
+/**
+ * Copyright (c) 2016-2018 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import { Component, ContentChild, OnDestroy, Optional } from '@angular/core';
+import { SelectMultipleControlValueAccessor } from '@angular/forms';
+import { Subscription } from 'rxjs';
+
+import { IfErrorService } from '../common/if-error/if-error.service';
+import { NgControlService } from '../common/providers/ng-control.service';
+import { LayoutService } from '../common/providers/layout.service';
+import { DynamicWrapper } from '../../utils/host-wrapping/dynamic-wrapper';
+import { ControlIdService } from '../common/providers/control-id.service';
+import { ClrLabel } from '../common/label';
+import { ControlClassService } from '../common/providers/control-class.service';
+
+@Component({
+  selector: 'clr-select-container',
+  template: `    
+        <ng-content select="label"></ng-content>
+        <label *ngIf="!label && addGrid()"></label>
```

Comment:
`hasGrid` would be more appropriate compared to `addGrid`?

Reply:
This is the name I've given elsewhere, though its indicative of when the component needs to add the grid (horizontal and compact layouts only).